internal/bzlmod/go_deps.bzl

```diff
@@ -326,17 +338,8 @@ def _go_deps_impl(module_ctx):
             root_module_direct_dev_deps.pop(_repo_name(path), default = None)
             continue
 
-        go_repository(
-            name = module.repo_name,
-            importpath = path,
-            sum = _get_sum_from_module(path, module, sums),
-            replace = getattr(module, "replace", None),
-            version = "v" + module.raw_version,
-            build_directives = _get_directives(path, gazelle_overrides),
-            build_file_generation = _get_build_file_generation(path, gazelle_overrides),
-            patches = _get_patches(path, module_overrides),
-            patch_args = _get_patch_args(path, module_overrides),
-        )
+        if not _archive_override_applied_as_go_repository(path, module, archive_overrides, gazelle_overrides, module_overrides):
```

Comment:
I would find it more readable if the repo rule instantiations and the predicate that switches between the two were inlined here. You could then share common args via `**kwargs`, ensuring that the two instantiations stay in sync.

Reply:
Good suggestion! updated to inline the args that vary